tools/block-generator/generator/generate.go

```diff
@@ -887,285 +785,21 @@ func (g *generator) generateAssetTxnInternalHint(txType TxTypeID, round uint64,
-		// can't do anything else with the app if it doesn't exist, so must create it first!!!
-		return getAppTxType(kind, appTxTypeCreate), kind, appTxTypeCreate, 0, nil
-	}
-
-	if appTxType == appTxTypeOptin {
-		// pick a random app to optin:
-		appID := g.appSlice[kind][rand.Uint64()%numAppsForKind].appID
-
-		_, exists, optedIn := g.getAppData(true /* existing */, kind, senderIndex, appID)
-		if !exists {
-			return txType, kind, appTxType, appID, fmt.Errorf("should never happen! app %d of kind %s does not exist", appID, kind)
-		}
-
-		if optedIn {
-			// already optedin, so call the app instead:
-			return getAppTxType(kind, appTxTypeCall), kind, appTxTypeCall, appID, nil
-		}
-
-		_, _, optedInPending := g.getAppData(false /* pending */, kind, senderIndex, appID)
-		if optedInPending {
-			// about to get opted in, but can't optin twice or call yet, so create:
-			return getAppTxType(kind, appTxTypeCreate), kind, appTxTypeCreate, appID, nil
-		}
-		// not opted in or pending, so optin:
-		return txType, kind, appTxType, appID, nil
-	}
-
-	if appTxType != appTxTypeCall {
-		return "", 0, 0, 0, fmt.Errorf("unimplemented transaction type for app %s from %s", appTxType, txType)
-	}
-	// WLOG appTxTypeCall:
-
-	numAppsOptedin := uint64(len(g.accountAppOptins[kind][senderIndex]))
-	if numAppsOptedin == 0 {
-		// try again calling recursively but attempting to optin:
-		return g.getActualAppCall(getAppTxType(kind, appTxTypeOptin), senderIndex)
-	}
-	// WLOG appTxTypeCall with available optins:
-
-	appID := g.accountAppOptins[kind][senderIndex][rand.Uint64()%numAppsOptedin]
-	return txType, kind, appTxType, appID, nil
-}
-
 // ---- metric data recorders ----
 
 func track(id TxTypeID) (TxTypeID, time.Time) {
 	return id, time.Now()
 }
 
 func (g *generator) recordData(id TxTypeID, start time.Time) {
-	g.recordOccurrences(id, 1, start)
-}
-
-func (g *generator) recordOccurrences(id TxTypeID, count uint64, start time.Time) {
-	g.latestData[id] += count
+	g.latestData[id]++
 	data := g.reportData[id]
-	data.GenerationCount += count
+	data.GenerationCount += 1
 	data.GenerationTime += time.Since(start)
 	g.reportData[id] = data
 }
 
-func (g *generator) countAndRecordEffects(id TxTypeID, start time.Time) (uint64, error) {
-	g.recordData(id, start) // this may be a bug!!!
-	count := uint64(1)
-	if consequences, ok := effects[id]; ok {
-		for _, effect := range consequences {
-			count += effect.count
-			g.recordOccurrences(effect.txType, effect.count, start)
-		}
-		return count, nil
-	}
-	return 1, fmt.Errorf("no effects for TxTypeId %v", id)
-}
-
-// ---- miscellaneous ----
-
-func (g *generator) txnForRound(round uint64) uint64 {
-	// There are no transactions in the 0th round
-	if round == 0 {
-		return 0
-	}
-	return g.config.TxnPerBlock
-}
-
-// startRound updates the generator's txnCounter based on the latest block header.
-// It is assumed that g.round has already been incremented in finishRound()
-func (g *generator) startRound() error {
```

Comment:
This was moved to `generator_ledger.go` and modified to set `g.txnCounter` even for round 0